versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5846 @@
+
+If a workflow has an optional input, its value may or may not be specified in the JSON input. It is also valid to explicitly set the value of an optional input to be undefined using JSON `null`.
+
+For example, given this workflow:
+
+```wdl
+workflow foo {
+  input {
+    File? x
+    Int? y = 5
+  }
+}
+```
+
+The following would all be valid JSON inputs:
+
+```json
+# no input
+{}
+
+# only x
+{
+  "x": 100
+}
+
+# only y
+{
+  "x": null,
+  "y": "/path/to/file"
+}
+
+# x and y
+{
+  "x": 1000,
+  "y": "/path/to/file"
+}
+
+# override y default and set it to None
+{
+  "y": null
+}
+```
+
+## JSON Output Format
+
+The outputs from a workflow invocation may be specified as a single JSON object that contains one member for each top-level workflow output; sub-workflow and task outputs are not provided. The name of the object member is the [fully-qualified name](#fully-qualified-names--namespaced-identifiers) of the output parameter, and the value is the [serialized form]() of the WDL value.
+
+Every WDL implementation must support the ability to output this standard output. It is suggested that WDL implementations make the standard format be the default output format.
+
+For example, given this workflow:
+
+```wdl
+workflow example {
+  ...
+  output {
+    String foo = cafeteria.inn
+    File analysis_results = analysis.results
+    Int read_count = readcounter.result
+    Float kessel_run_parsecs = trip_to_space.distance
+    Boolean sample_swap_detected = array_concordance.concordant
+    Array[File] sample_variants = variant_calling.vcfs
+    Map[String, Int] droids = escape_pod.cargo
+  }
+}
+```
+
+The output JSON will look like:
+
+```json
+{
+  "example.foo": "bar",
+  "example.analysis_results": "/path/to/my/analysis/results.txt",
+  "example.read_count": 50157187,
+  "example.kessel_run_parsecs": 11.98,
+  "example.sample_swap_detected": false,
+  "example.sample_variants": ["/data/patient1.vcf", "/data/patient2.vcf"],
+  "example.droids": {"C": 3, "D": 2, "P": 0, "R": 2}
+}
+```
+
+It is recommended (but not required) that JSON outputs be "pretty printed" to be more human-readable.
+
+## Specifying / Overriding Runtime Attributes
+
+In addition to specifying workflow input in JSON, the user can also specify (or override) runtime attributes. Input values for standardized runtime attributes must adhere to the [supported types and formats](#runtime-section). Runtime values provided as inputs always supersede values supplied directly in the WDL. Any runtime attributes that are not supported by the execution engine are ignored.
+
+To differentiate runtime attributes from task inputs, the `runtime` namespace is added after the task name. For example:
+
+```json
+{
+  "wf.t1.runtime.memory": "16 GB",
+  "wf.t2.runtime.cpu": 2,
+  "wf.t2.runtime.disks": "100",
+  "wf.t2.runtime.container": "mycontainer:latest"
+}
+```
+
+## JSON Serialization of WDL Types
+
+### Primitive Types
+
+All WDL types serialize naturally to JSON values:
```

Comment:
All **primitive** WDL types serialize naturally to JSON values:

Reply:
good catch